src/pool/pool.service.ts

```diff
@@ -0,0 +1,160 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { InjectRepository } from '@nestjs/typeorm';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import ApolloClient from 'apollo-client';
+import { HttpLink } from 'apollo-link-http';
+
+import gql from 'graphql-tag';
+import { Repository } from 'typeorm';
+import fetch from 'cross-fetch';
+
+import { Pool } from './pool.entity';
+import Big from 'big.js';
+
+export const createClient = (uri: string) =>
+  new ApolloClient({
+    link: new HttpLink({ uri, fetch: fetch }),
+    cache: new InMemoryCache(),
+  });
+
+export const HOUR_IN_SECONDS = 60 * 60;
+export const HOURS_IN_DAY = 24;
+
+export const createQuery = (
+  blockTimestamp_lte: string | number,
+  blockTimestamp_gte: string | number,
+) => gql`
+  query Swaps {
+    swaps(
+      first: 1000,
```

Comment:
Please implement basic pagination here. Even if you expect a 1000 of records to be enough for eternity.